pipenv/cli.py

```diff
@@ -137,10 +138,13 @@ def add_to_path(p):
 @background.task
 def check_for_updates():
     """Background thread -- beautiful, isn't it?"""
+    if not need_update_check():
```

Comment:
It's gonna be more work for me to fork and submit a PR to your branch @techalchemy.  All I was going to do, pending the outcome of your tests, was to remove statement to be outside of the thread.  This would move some IO to the main cli but I don't think it will be an issue.

Reply:
oh yeah give me one sec and i will let you know, I was going to do the actual math on it but i'll eyeball and see if there is even a difference at a glance

Reply:
I'm pretty sure network IO will always be larger than thread spawn time. I'd suggest keeping network calls out of the main thread unless there's significant data showing otherwise.

Reply:
OK your approach is faster than mine by like ~0,03s on average so I'd say just move that into the cli as you were going to do:

```python
    if not update and need_update_check():
        # Spun off in background thread, not unlike magic.
        check_for_updates()
```

Reply:
@nateprewitt we were mainly talking about the disk IO associated with touching/opening a file pointer, i.e. just whether we should determine if we need to check for an update inside the background thread that will do the network call, or determine whether we need to check for an update before we even spawn the background thread (thereby preventing it from spawning)